src/tzdata/compile.jl

```diff
@@ -694,7 +694,7 @@ function compile(tz_source::TZSource, dest_dir::AbstractString; kwargs...)
     results = compile(tz_source; kwargs...)
 
     isdir(dest_dir) || error("Destination directory doesn't exist")
-    empty!(TIME_ZONE_CACHE)
+    empty!(default_tz_cache())
```

Comment:
Re-running compile should invalidate the caches on all threads. I'm wondering though if this is still thread-safe then...
```suggestion
    TimeZones._reset_tz_cache()
```

Reply:
It seems like this is the sole outstanding comment, and otherwise this pull request is in good shape? :)

Reply:
Yeah, i don't know what to do about this... @omus is right that this would no longer be thread safe, and i don't know how to make it thread-safe without _also_ introducing a lock around every access to the tz_cache, which defeats the whole point of this PR, and in which case something more like #345 might make more sense.

@omus: Can you explain more about why this function is invalidating the existing cache? In what way does it make the previously computed values invalid?

Reply:
Hi gentlemen! What's the path forward here? @omus, any thoughts on Nathan's question just above? :)

Reply:
I've asked about this line, here:
https://github.com/JuliaTime/TimeZones.jl/commit/182b6599a2dfede2114e74ce4817599ee086935a#r55083604

Reply:
Done. Thanks!